File: scala/problem1.scala
object Main {
  def main(args: Array[String]) {
    val n = if (args.isEmpty) 1000 else args(0).toInt
    val result = (1 until n).filter(x => x % 3 == 0 || x % 5 == 0).sum
    println(result)
  }
}

File: erlang/problem1.erl
#!/usr/bin/env escript

main(_) ->
  solve().

solve() ->
  RESULT = lists:sum([X || X <- lists:seq(1,999), (X rem 3 == 0) or (X rem 5 == 0)]),
  io:format("~p~n", [RESULT]).
